modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-select/src/main/resources/META-INF/resources/select_search.js

```diff
@@ -0,0 +1,143 @@
+AUI.add(
+	'liferay-ddm-form-field-select-search-support',
+	function(A) {
+		var AArray = A.Array;
+
+		var CSS_SEARCH_INPUT = 'drop-chosen-search';
+
+		var Lang = A.Lang;
+
+		var SoyTemplateUtil = Liferay.DDM.SoyTemplateUtil;
+
+		var SelectFieldSearchSupport = function() {};
+
+		SelectFieldSearchSupport.prototype = {
+			initializer: function() {
+				var instance = this;
+
+				instance._eventHandlers.push(
+					instance.bindContainerEvent('input', A.debounce(instance._afterStartSearching, 400), '.' + CSS_SEARCH_INPUT),
+					instance.after('closeList', A.bind('_afterCloseList', instance)),
+					instance.after(A.bind('_afterOpenList', instance), instance, 'openList')
+				);
+			},
+
+			clearFilter: function() {
+				var instance = this;
+
+				var searchInputNode = instance._getInputSearch();
+
+				var options = instance.get('context').options;
+
+				instance._renderList(options);
+
+				searchInputNode.val('');
+			},
+
+			_afterCloseList: function() {
+				var instance = this;
+
+				instance.clearFilter();
+			},
+
+			_afterOpenList: function() {
+				var instance = this;
+
+				var searchInputNode = instance._getInputSearch();
+
+				searchInputNode.focus();
+			},
+
+			_afterStartSearching: function(event) {
+				var instance = this;
+
+				var target = event.target;
+
+				var term = Lang.trim(target.get('value')).toLowerCase();
+
+				var filteredOptions = AArray.filter(
+					instance.get('options'),
+					function(option) {
+						return instance._findString(option.label, term) >= 0;
+					}
+				);
+
+				instance._renderList(filteredOptions);
+
+				instance._visitDOMListItens(
+					A.bind(instance._applyFilterStyleOnItem, instance, term)
+				);
+			},
+
+			_applyFilterStyleOnItem: function(value, item) {
+				var instance = this;
+
+				var content = item.getContent();
+
+				item.setContent(instance._injectHighLight(content, value));
+			},
+
+			_findString: function(fullString, term) {
+				return fullString.toLowerCase().indexOf(term);
+			},
+
+			_getInputSearch: function() {
+				var instance = this;
+
+				return instance.get('container').one('.' + CSS_SEARCH_INPUT);
+			},
+
+			_getTemplate: function(context) {
+				var instance = this;
+
+				var renderer = SoyTemplateUtil.getTemplateRenderer('ddm.select_options');
+
+				return renderer(context);
+			},
+
+			_injectHighLight: function(string, term) {
```

Comment:
I suggest trying to look into using http://yuilibrary.com/yui/docs/api/files/highlight_js_highlight-accentfold.js.html#l21 because (if I remember correctly) it already handles a lot of edge cases like Japanese characters and letters with accent. Can you look into that?

Thanks,